programs/mango-v4/src/instructions/openbook_v2_place_order.rs

```diff
@@ -0,0 +1,565 @@
+    // value go through the roof, reducing available init margin for other users.
+    let band_threshold = openbook_market.oracle_price_band();
+    if new_order_on_book && band_threshold != f32::MAX {
+        let (base_oracle, quote_oracle) = match order.side {
+            OpenbookV2Side::Bid => (&receiver_bank_oracle, &payer_bank_oracle),
+            OpenbookV2Side::Ask => (&payer_bank_oracle, &receiver_bank_oracle),
+        };
+        let base_oracle_f64 = base_oracle.to_num::<f64>();
+        let quote_oracle_f64 = quote_oracle.to_num::<f64>();
+        // this has the same units as base_oracle: USD per BASE; limit_price is in QUOTE per BASE
+        let limit_price_in_dollar = limit_price * quote_oracle_f64;
+        let band_factor = 1.0 + band_threshold as f64;
+        match order.side {
+            OpenbookV2Side::Bid => {
+                require_msg_typed!(
+                    limit_price_in_dollar * band_factor >= base_oracle_f64,
+                    MangoError::SpotPriceBandExceeded,
+                    "bid price {} must be larger than {} ({}% of oracle)",
+                    limit_price,
+                    base_oracle_f64 / (quote_oracle_f64 * band_factor),
+                    (100.0 / band_factor) as u64,
+                );
+            }
+            OpenbookV2Side::Ask => {
+                require_msg_typed!(
+                    limit_price_in_dollar <= base_oracle_f64 * band_factor,
+                    MangoError::SpotPriceBandExceeded,
+                    "ask price {} must be smaller than {} ({}% of oracle)",
+                    limit_price,
+                    base_oracle_f64 * band_factor / quote_oracle_f64,
+                    (100.0 * band_factor) as u64,
+                );
+            }
+        }
+    }
+
+    // Health cache updates for the changed account state
+    let receiver_bank = receiver_bank_ai.load::<Bank>()?;
+    // update scaled weights for receiver bank
+    health_cache.adjust_token_balance(&receiver_bank, I80F48::ZERO)?;
+    vault_difference.adjust_health_cache_token_balance(&mut health_cache, &payer_bank)?;
+    let openbook_account = account.openbook_v2_orders(openbook_market.market_index)?;
+    oo_difference.recompute_health_cache_openbook_v2_state(
+        &mut health_cache,
+        &openbook_account,
+        &open_orders,
+    )?;
+
+    // Check the receiver's reduce only flag.
+    //
+    // Note that all orders on the book executing can still cause a net deposit. That's because
+    // the total openbook_v2 potential amount assumes all reserved amounts convert at the current
+    // oracle price.
+    if receiver_bank_reduce_only {
+        let balance = health_cache.token_info(receiver_token_index)?.balance_spot;
+        let potential =
+            health_cache.total_spot_potential(HealthType::Maint, receiver_token_index)?; // todo-pan: split potential into serum and openbook
```

Comment:
hmm, shouldn't it stay combined? we don't want to cause deposits even if all orders over all relevant markets execute